### ISSUE TITLE
Java 7/Crouton/ARM and Minecraft

### ISSUE THREAD

**Issue by mlevie** [dnschneid/crouton#1484]
I'm so sorry if this is the wrong forum...please feel free to redirect me. I saw a similar question elsewhere on this forum!

Samsung Chromebook XE303C12
KDE Linux 3.8.11 (32-bit)
Java version 1.7.0_60

I'm not able to launch Minecraft. I am 99% sure this has to do with my Chromebook being one of the ARM-based Samsung models. I totally realize this is Not Recommended, but for my nine-year-old's sake I'm trying to do it anyway.

I found a thread here which said that Java 7 as distributed doesn't support ARM processors, so I downloaded a version of 1.7.0_60 that said it supports hard float on ARMs. But I'm still getting the same kind of errors.

If you have any ideas, please let me know!

In the Game Output you get:

OpenJDK Zero VM warning: You have loaded library /home/baserty/.minecraft/versions/1.8.3/1.8.3-natives-44326013494133/liblwjgl64.so which might have disabled stack guard. The VM will try to fix the stack guard now.
It's highly recommended that you fix the library with 'execstack -c <libfile>', or link it with '-z noexecstack'.
Exception in thread "main" java.lang.UnsatisfiedLinkError: /home/baserty/.minecraft/versions/1.8.3/1.8.3-natives-44326013494133/liblwjgl64.so: /home/baserty/.minecraft/versions/1.8.3/1.8.3-natives-44326013494133/liblwjgl64.so: cannot open shared object file: No such file or directory (Possible cause: can't load IA 32-bit .so on a ARM-bit platform)
at java.lang.ClassLoader$NativeLibrary.load(Native Method)
at java.lang.ClassLoader.loadLibrary1(ClassLoader.java:1965)
at java.lang.ClassLoader.loadLibrary0(ClassLoader.java:1890)
at java.lang.ClassLoader.loadLibrary(ClassLoader.java:1880)
at java.lang.Runtime.loadLibrary0(Runtime.java:849)
at java.lang.System.loadLibrary(System.java:1088)
at org.lwjgl.Sys$1.run(Sys.java:72)
at java.security.AccessController.doPrivileged(Native Method)
at org.lwjgl.Sys.doLoadLibrary(Sys.java:66)
at org.lwjgl.Sys.loadLibrary(Sys.java:96)
at org.lwjgl.Sys.<clinit>(Sys.java:117)
at avf.J(SourceFile:2524)
at net.minecraft.client.main.Main.main(SourceFile:41)

**Comment by smibarber** [comment]
The particular problem you're having is due to Minecraft's dependency on some libraries that are built for x86 (`liblwjgl64.so`). But even getting past that, there won't be any 3D acceleration on ARM (at least with Mali graphics) so the game itself will be unplayable.

**Comment by mlevie** [comment]
So does that mean it's possible to get past that? I thought that was what I was hoping to achieve by installing the Java version for ARM, but are you saying that there's no potential replacement of this library for ARM processors?

To be honest, I've spent enough time on this that even if the game were unplayable, it would be worth getting it to run.

**Comment by smibarber** [comment]
It's definitely possible, and it looks like it's been done before [here](http://forum.lwjgl.org/index.php?topic=5494.0). I can't vouch for the accuracy of those instructions though ;)

**Comment by mlevie** [comment]
OK! I got it working and can positively confirm that, yeah, it's totally unplayable on an ARM machine. However I'm still a hero to my son for getting it to run so I am in your debt. Plus I learned a ton.

Just in case anyone else decides to attempt this...although why you would, now that I've confirmed it will run at maybe one frame a second, is beyond me...

1) Use <a href="http://itechtriad.com/articles/2014/7/21/how-to-install-minecraft-on-a-chromebook">this  article</a> to install KDE Linux on the Chromebook.

2) Go <a href="http://rogerallen.github.io/jetson/2014/07/31/minecraft-on-jetson-tk1/">here</a> to build the ARM-compliant version of the liblwjgl.so and libopenal.so libraries.

3) When you get to Preliminary Run of Minecraft, use version 1.6.4. The most recent versions only seem to use downloaded versions of LWJGL so there's not really any local version to replace.

4) Install zip: <b>sudo get-apt install zip</b>. It's not a part of KDE Linux by default.

4) Use <a href="https://gist.github.com/Coderlane/079e3b77369a6a1d2b01">this script</a> to copy those libraries into the .jar file where the liblwjgl.so and libopenal.so libraries are stored. Note that you will need to look in ~/.minecraft/libraries/org/lwjgl/lwjgl/lwjgl-platform to check the version. The script says 2.9.1 but my version was using 2.9.0.
